x-pack/plugins/security_solution/public/flyout/document_details/left/components/add_note.tsx

```diff
@@ -51,29 +76,40 @@ export interface AddNewNoteProps {
 }
 
 /**
- * Renders a markdown editor and a add button to create new notes
+ * Renders a markdown editor and an add button to create new notes.
+ * The checkbox is automatically checked if the flyout is opened from a timeline and that timeline is saved. It is disabled if the flyout is NOT opened from a timeline.
  */
 export const AddNote = memo(({ eventId }: AddNewNoteProps) => {
   const dispatch = useDispatch();
   const { addError: addErrorToast } = useAppToasts();
   const [editorValue, setEditorValue] = useState('');
 
+  const activeTimeline = useSelector((state: State) =>
+    timelineSelectors.selectTimelineById(state, TimelineId.active)
+  );
+
+  // if the flyout is open from a timeline and that timeline is saved, we automatically check the checkbox to associate the note to it
+  const isTimelineFlyout = useIsTimelineFlyoutOpen();
+  const [checked, setChecked] = useState(isTimelineFlyout && activeTimeline.savedObjectId != null);
```

Comment:
should this be `activeTimeline?.savedObjectId`?

Reply:
it doesn't have to, I think in a normal user environment `activeTimeline` is always defined. I used the `?` for the other places` because the unit tests were failing as for unit test we use a global state that has `TimelineId.test` instead of TimelineId`active`..

I think the tests are wrong but this PR isn't the place to update that...

Does it make sense?

Reply:
👍🏾  Yea, makes sense. `TimelineId.Active` shouldn't be static 😞